profiler/src/main/java/eu/clarin/switchboard/profiler/api/Profile.java

```diff
@@ -0,0 +1,187 @@
+        return mediaType;
+    }
+
+    public String getLanguage() {
+        return language;
+    }
+
+    public boolean isMediaType(String mediaType) {
+        return this.mediaType.equalsIgnoreCase(mediaType);
+    }
+
+    public Map<String, String> getFeatures() {
+        return features;
+    }
+
+    public boolean isXmlMediaType() {
+        // mediatype format is: type "/" [tree "."] subtype ["+" suffix]* [";" parameter]*
+        // 1. remove parameter suffix
+        int formatStartIndex = mediaType.indexOf(';');
+        if (formatStartIndex > 0) {
+            mediaType = mediaType.substring(0, formatStartIndex);
+        }
+        // 2. split in type and subtypes
+        String[] typeSubtype = mediaType.split("/");
+        if (typeSubtype.length == 2) {
+            String subtype = typeSubtype[1];
+            // 3. find and remove tree name
+            int i = subtype.indexOf(".");
+            if (i >= 0) {
+                subtype = typeSubtype[1].substring(i + 1);
+            }
+            // 4. find suffix
+            String suffix = "";
+            i = subtype.lastIndexOf("+");
+            if (i >= 0) {
+                suffix = subtype.substring(i + 1);
+                subtype = subtype.substring(0, i);
+            }
+            return "xml".equalsIgnoreCase(subtype) || "xml".equalsIgnoreCase(suffix);
+        }
+        return false;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Profile profile = (Profile) o;
+        return Objects.equals(confidence, profile.confidence) &&
+                Objects.equals(mediaType, profile.mediaType) &&
+                Objects.equals(language, profile.language) &&
+                Objects.equals(features, profile.features);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(mediaType, language, features);
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder buf = new StringBuilder("Profile { confidence=");
+        buf.append(confidence).append("; mediaType=").append(mediaType);
+        if (language != null) {
+            buf.append("; language=").append(language);
+        }
+        for (String k : features.keySet()) {
+            buf.append("; ").append(k);
+            String v = features.get(k);
+            if (v != null) {
+                buf.append("=").append(v);
+            }
+        }
+        buf.append(" }");
+        return buf.toString();
+    }
+
+    @Override
+    public int compareTo(@NotNull Profile o) {
+        int x = confidence.compareTo(o.confidence);
+        if (x == 0 && mediaType != null) {
+            x = mediaType.compareToIgnoreCase(o.mediaType);
+        }
+        if (x == 0 && language != null) {
+            x = language.compareToIgnoreCase(o.language);
+        }
+        return x;
+    }
+
+    public static class Builder {
```

Comment:
I like the builder pattern, but isn't one of the main points that it allows for construction of an immutable object? So the 'Profile' class could be changed to only provide immutable instances?